src/Disks/ObjectStorages/AzureBlobStorage/AzureBlobStorageAuth.cpp

```diff
@@ -58,28 +86,96 @@ void validateContainerName(const String & container_name)
 
 AzureBlobStorageEndpoint processAzureBlobStorageEndpoint(const Poco::Util::AbstractConfiguration & config, const String & config_prefix)
 {
-    std::string storage_url;
-    if (config.has(config_prefix + ".storage_account_url"))
+    String storage_url;
+    String account_name;
+    String container_name;
+    String prefix;
+    if (config.has(config_prefix + ".endpoint"))
+    {
+        String endpoint = config.getString(config_prefix + ".endpoint");
+
+        /// For some authentication methods account name is not present in the endpoint
+        /// 'endpoint_contains_account_name' bool is used to understand how to split the endpoint (default : true)
```

Comment:
`For some authentication methods account name is not present in the endpoint`

is it possible to make it `false` by default ? Looks like it's used w/ azure blob store emulator ?

Reply:
>  Looks like it's used only w/ azure blob store emulator ?
Sorry I didnt get this. 

I have updated the default value of `endpoint_contains_account_name` to false.